stdlib/p7zip_jll/src/p7zip_jll.jl

```diff
@@ -88,6 +87,7 @@ function __init__()
     append!(LIBPATH_list, [joinpath(Sys.BINDIR, Base.LIBDIR, "julia"), joinpath(Sys.BINDIR, Base.LIBDIR)])
     LIBPATH[] = join(LIBPATH_list, pathsep)
 end
+precompile(Tuple{typeof(__init__)})
```

Comment:
@vtjnash I seem to remember you telling me in the past that this is either unnecessary or undesirable.  Is that still the case?

Reply:
That is still the case

Reply:
Ok. I'll remove them. It'd be helpful to understand why, because on face value (based on `--trace-compile`)it avoids compilation during load and the reason why it's unnecessary or undesirable isn't clear to me

Reply:
The removal of just the 
```
Base.Experimental.@compiler_options compile=min optimize=0 infer=false
```
appears to avoid the inits in `--trace-compile` during loading Pkg. So I'll take it that it's more that it's unnecessary.